src/main/default/lwc/voteTracker/__test__/voteTracker.test.js

```diff
@@ -0,0 +1,340 @@
+
+      expect(negativeButtonNode.state).toBe(expectedNegativeState);
+      expect(positiveButtonNode.state).toBe(expectedPositiveState);
+    });
+
+    it('should switch the component to the final text', async () => {
+      const element = createTestComponent();
+
+      await flushPromises();
+      const positiveButtonNode = element.shadowRoot.querySelectorAll(
+        'c-vote-button'
+      )[0];
+
+      const clickEvent = new CustomEvent('click');
+      await positiveButtonNode.dispatchEvent(clickEvent);
+
+      jest.runAllTimers();
+
+      expect(element.finalState).toBe(true);
+    });
+
+    it('should display the correct final text', async () => {
+      const element = createTestComponent();
+      const expectedText = 'Final Text';
+      element.finalText = expectedText;
+
+      await flushPromises();
+      const positiveButtonNode = element.shadowRoot.querySelectorAll(
+        'c-vote-button'
+      )[0];
+
+      const clickEvent = new CustomEvent('click');
+      await positiveButtonNode.dispatchEvent(clickEvent);
+
+      await jest.runAllTimers();
+      const finalTextNode = element.shadowRoot.querySelector(
+        'h3.slds-text-title_bold'
+      );
+
+      expect(finalTextNode).not.toBeNull();
+      expect(finalTextNode.textContent).toBe(expectedText);
+    });
+
+    it('should display the correct default localized final text', async () => {
+      const element = createTestComponent();
+      const expectedText = 'Thank you for the feedback!';
+
+      await flushPromises();
+      const positiveButtonNode = element.shadowRoot.querySelectorAll(
+        'c-vote-button'
+      )[0];
+
+      const clickEvent = new CustomEvent('click');
+      await positiveButtonNode.dispatchEvent(clickEvent);
+
+      await jest.runAllTimers();
+      const finalTextNode = element.shadowRoot.querySelector(
+        'h3.slds-text-title_bold'
+      );
+
+      expect(finalTextNode).not.toBeNull();
+      expect(finalTextNode.textContent).toBe(expectedText);
+    });
+
+    it('should show the abandon button request', async () => {
+      const element = createTestComponent();
+      element.size = 'big';
+
+      await flushPromises();
+      const positiveButtonNode = element.shadowRoot.querySelectorAll(
+        'c-vote-button'
+      )[0];
+
+      const clickEvent = new CustomEvent('click');
+      await positiveButtonNode.dispatchEvent(clickEvent);
+
+      await jest.runAllTimers();
+      const buttonNode = element.shadowRoot.querySelector('lightning-button');
+
+      expect(buttonNode).not.toBeNull();
+    });
+  });
+  describe('when the negative voteButton is clicked', () => {
+    it('should change the state of the voteButtons', async () => {
+      const element = createTestComponent();
+      const expectedPositiveState = 'neutral';
+      const expectedNegativeState = 'selected';
+
+      await flushPromises();
+      const positiveButtonNode = element.shadowRoot.querySelectorAll(
+        'c-vote-button'
```

Comment:
You could add marker classes to the positive and  negative buttons to be able to select them more easily. It might be more reliable than indexes over time.

`vote-tracker__positive-button` / `vote-tracker__negative-button`